networking/configuring-a-custom-pki.adoc

```diff
@@ -0,0 +1,57 @@
+[id="configuring-a-custom-pki"]
+= Configuring a custom PKI
+include::modules/common-attributes.adoc[]
+:context: configuring-a-custom-pki
+
+toc::[]
+
+Some platform components (i.e., web console) use Routes for communication and,
+therefore, must trust other components' certificates to interact with them. If
+you're using a custom public key infrastructure (PKI), you must configure it so
+its privately signed CA certificates are recognized across the cluster.
+
+You can leverage the Proxy API to add cluster-wide trusted CA certificates. You
+must do this either during installation *or* at runtime.
+
+* During *installation*, configure the cluster-wide proxy. You must define your
+privately signed CA certificates in the `install-config.yaml` file's
+`additionalTrustBundle` setting.
++
+The installation program generates a ConfigMap that is named `user-ca-bundle`
+that contains the additional CA certificates you defined. The Cluster Network
+Operator then creates a `trusted-ca-bundle` ConfigMap that merges these CA
+certificates with the {op-system-first} trust bundle; this ConfigMap is
+referenced in the Proxy object's `trustedCA` field.
+
+* At *runtime*, modify the default Proxy object to include your privately signed
+CA certificates (part of cluster's proxy enablement workflow). This involves
+creating a ConfigMap that contains the privately signed CA certificates that
+should be trusted by the cluster, and then modifying the proxy resource with the
+`trustedCA` referencing the privately signed certificates' ConfigMap.
+
+[NOTE]
+====
+The installer configuration's `additionalTrustBundle` field and the proxy
+resource's `trustedCA` field are used to manage the cluster-wide trust bundle;
+`additionalTrustBundle` is used at install time and the proxy's `trustedCA` is
+used at runtime.
+
+The `trustedCA` field is a reference to a `ConfigMap` containing the custom
+certificate and key pair used by the cluster component.
+====
+
+include::modules/installation-configure-proxy.adoc[leveloffset=+1]
```

Comment:
So, the way this is laid out now kind of makes it look like you need to do both of these procedures in the modules.

I'd consider linking to each of these from earlier in this assembly where you talk about configuring during installation or at runtime.

Reply:
@bergerhoffer yeah, I was struggling with that. I was unsure whether it was best practice to link to a module from within its own assembly. I can't think of a reason why that wouldn't work, since we know the module will always be available from within the article.

I'll go ahead and provide links; I agree that would be much better.